src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultsOfEachAction = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (true) {
+      case action.type === 'addProperties':
```

Comment:
```suggestion
    switch (action.type) {
      case 'addProperties':
```
You can write the key of an object in the condition in `switch`, and check the value of the key in `case`.